src/histolab/tiler.py

```diff
@@ -276,6 +305,7 @@ def extract(
         level = logging.getLevelName(log_level)
         logger.setLevel(level)
         self._validate_level(slide)
+        self._set_proper_tile_size_and_overlap(slide)
```

Comment:
I think that this line should be:
```
if self.mpp is None:
    tile_size = self.tile_size
    sf = 1.0
else:
    sf = self.mpp / slide.base_mpp
    tile_size = tuple(int(j * sf) for j in self.tile_size)
self.tile_size =  tile_size
self.pixel_overlap = int(sf * self.pixel_overlap)
```

and then you will have 2 setters like:

```
@pixel_overalp.setter
def pixel_overlap(self, value):
    self.pixel_overlap = value

@tile_size.setter
def tile_size(self, value):
   self._tile_size = value
```

This way you could get rid of `_set_proper_tile_size_and_overlap` and `_get_proper_tile_size` and respect one of the python zen rule `Explicit is better than implicit` and `Beautiful is better than ugly` 😉

Reply:
@ernestoarbitrio Actually I prefer to keep it the way it is now to avoid repeating code too much to reduce chances of bugs. The ``_get_proper_tile_size()`` method is in the parent ``Tiler`` module and used by both ``GridTiler`` and ``ScoreTiler`` .. if I get rid of it I'll have to repeat code which makes it harder to maintain and more bug-prone. Also, there's already an existing ``@tile_size.setter``, so this would conflict with adding a new tile size setter. Note that the method ``_set_proper_tile_size_and_overlap`` method is used by both ``GridTiler`` and inherited and used by ``ScoreTiler`` so writing it would will also result in code repetition which could be bug-prone. Another reason to keep ``_set_proper_tile_size_and_overlap`` explicitly is that it can be unit tested as an independent functionality so if something is wrong with the any ``Tiler`` when ``mpp is not None``, we can know if the problem is caused by scale factor differences or some other problem in the ``extract`` method.

Reply:
Ok I can buy this for now, but we should think at a better implementation in the future